src/main/kotlin/com/terraformation/backend/documentproducer/db/variable/AllVariableCsvVariable.kt

```diff
@@ -0,0 +1,110 @@
+    val minValue: BigDecimal?,
+    /** Column 9/I - Maximum value, if any (for Number) */
+    val maxValue: BigDecimal?,
+    /** Column 10/J - Decimal places (for Number) */
+    val decimalPlaces: Int?,
+    /** Column 11/K - Table style (for Table) */
+    val tableStyle: VariableTableStyle?,
+    /** Column 12/L - Header? (for table columns) */
+    val isHeader: Boolean,
+    /** Column 13/M - Notes */
+    val notes: String?,
+    /** Column 14/N - Deliverable ID */
+    val deliverableId: DeliverableId?,
+    /** Column 15/O - Deliverable Question */
+    val deliverableQuestion: String?,
+    /** Column 16/P - Dependency - Variable Stable ID */
+    val dependencyVariableStableId: String?,
+    /** Column 17/Q - Dependency - Condition */
+    val dependencyCondition: DependencyCondition?,
+    /** Column 18/R - Dependency - Value */
+    val dependencyValue: String?,
+    /** Column 19/S - Internal Only */
+    val internalOnly: Boolean,
+
+    /** Position in the sheet, recorded against the variable manifest entry */
+    val position: Int,
+    /** The path to the variable within the hierarchy */
+    var variablePath: String,
+    /** The path to the parent variable within the hierarchy */
+    var parentPath: String?,
+    /** The base variable ID */
+    var variableId: VariableId? = null,
+    /** If this variable replaces one from an earlier manifest, that variable's ID. */
+    var replacesVariableId: VariableId? = null,
+) {
+  fun mapToVariablesRow() =
+      VariablesRow(
+          deliverableId = deliverableId,
+          deliverableQuestion = deliverableQuestion,
+          dependencyVariableStableId = dependencyVariableStableId,
+          dependencyConditionId = dependencyCondition,
+          dependencyValue = dependencyValue,
+          description = description,
+          id = variableId,
+          internalOnly = internalOnly,
+          isList = isList,
+          name = name,
+          replacesVariableId = replacesVariableId,
+          stableId = stableId,
+          variableTypeId = dataType.variableType,
+      )
+}
+
+enum class AllVariableCsvVariableType(val value: String, val variableType: VariableType) {
+  Number("Number", VariableType.Number),
+  SingleLine("Text (single-line)", VariableType.Text),
+  MultiLine("Text (multi-line)", VariableType.Text),
+  SingleSelect("Select (single)", VariableType.Select),
+  MultiSelect("Select (multiple)", VariableType.Select),
+  Date("Date", VariableType.Date),
+  Link("Link", VariableType.Link),
+  Image("Image", VariableType.Image),
+  Table("Table", VariableType.Table);
+
+  companion object {
+    private val byValue = entries.associateBy { it.value }
+
+    fun create(input: String): AllVariableCsvVariableType =
+        byValue[input] ?: throw IllegalArgumentException(LOCALIZED_ERROR_KEY_UNKNOWN_DATA_TYPE)
+  }
+}
+
+enum class HierarchicalVariableType(val value: String) {
+  Table("Table"),
+  Section("Section")
+}
```

Comment:
Maybe this should move to its own top-level file since it's used by both the all-variables and the manifest importer.